esmvalcore/preprocessor/_mask.py

```diff
@@ -52,11 +52,11 @@ def _get_fx_mask(fx_data, fx_option, mask_type):
 def _apply_fx_mask(fx_mask, var_data):
     """Apply the fx data extracted mask on the actual processed data."""
     # Apply mask across
-    if np.ma.is_masked(var_data):
-        fx_mask |= var_data.mask
-
-    # Build the new masked data
-    var_data = np.ma.array(var_data, mask=fx_mask, fill_value=1e+20)
+    # Then apply the mask (with getmaskarray)
+    old_mask = da.ma.getmaskarray(var_data)
+    mask = old_mask | fx_mask
+    var_data = da.ma.masked_array(var_data, mask=mask)
+    # maybe fill_value=1e+20
```

Comment:
```suggestion
```

Reply:
nah I'd keep that in, if we don't force it in code, then let's see if it ever hits us back, that's a good pointer where to plug it back in